lib/reversetunnel/seek/seek.go

```diff
@@ -389,18 +389,22 @@ func (p *proxyGroup) releaseProxy(t time.Time, principals ...string) (ok bool) {
 }
 
 func (p *proxyGroup) resolveName(principals []string) string {
-	const uuidSize int = 36
+	// check if we're alsready using one of these principals
 	for _, name := range principals {
-		if len(name) >= uuidSize {
-			if uuid.Parse(name[:uuidSize]) != nil {
-				return name[:uuidSize]
-			}
-		}
 		if _, ok := p.states[name]; ok {
 			return name
 		}
 	}
-	return principals[0]
+	// default to using the first principal
+	name := principals[0]
+	// if we have a `.<cluster-name>` suffix, remove it.
+	if strings.HasSuffix(name, p.id.Cluster) {
```

Comment:
`strings.TrimSuffix` will return `s` unchanged if the suffix does not exist: https://golang.org/pkg/strings/#TrimSuffix

What about this?

```go
name = strings.TrimSuffix(name, p.id.Cluster)
name = strings.TrimSuffix(name, ".")
```

Reply:
> What about this?
> 
> ```go
> name = strings.TrimSuffix(name, p.id.Cluster)
> name = strings.TrimSuffix(name, ".")
> ```

This may break names which coincidentally end in clustername (e.g. if clustername is `beef` and hostname is `deadbeef`, hostname would be incorrectly trimmed to `dead`).

The use of `HasSuffix` is intended to replicate the behavior of something like this:

```go
strings.TrimSuffix(name, "." + p.id.Cluster)
``` 
...but without unnecessary allocations.

Reply:
Got it.